### PR TITLE
Animations

### PR DESCRIPTION
**Pull request by imf**

### CODE REVIEW

File: Say Their Names/App.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+//  Created by Jacob Martin on 6/29/20.
+//
+
+import Foundation
+import ComposableArchitecture
+import Combine
+
+//MARK: - State
+
+enum AppViewState: Equatable {
```
Comment (j-j-m) [inline Say Their Names/App.swift:14]: These case names and the struct can change. kinda came up with them quickly. its seems like more of a view mode than a state. in any case kind of moot at the moment

Comment (imf) [reply Say Their Names/App.swift:14]: Yes makes sense! They'll tell us what they want to be called.

File: Say Their Names/App.swift [hunk 2]
```diff
@@ -43,16 +54,25 @@ let appReducer: Reducer<AppState, AppAction, AppEnvironment> =
         state, action, environment in
         switch action {
         case .loadIncedents:
-            return environment
-                .incedents()
-                .receive(on: environment.mainQueue)
-                .catchToEffect()
-                .map(AppAction.incedentsResponse)
+            return Effect(value: .startLoading)
```
Comment (imf) [inline Say Their Names/App.swift:57]: Oh fun!

File: Say Their Names/ContentView.swift [hunk 3]
```diff
@@ -19,20 +20,80 @@ struct ContentView: View {
...
+                    .frame(minHeight: 60)
+                    .transition(.scale)
             } else {
-                VStack {
-                    NameList(incidents: viewStore.incidents)
-                    Spacer()
-                    Button("Say their names!", action: {
-                        NameSayer.saySomeNames(incidents: self.viewStore.incidents, howMany: 20)
-                    })
+                Button("Load their names!", action: {
```
Comment (imf) [inline Say Their Names/ContentView.swift:44]: This probably should be automatic. But of course for now we can make it user-initiated.

File: Say Their Names/ContentView.swift [hunk 4]
```diff
@@ -19,20 +20,80 @@ struct ContentView: View {
...
+        HStack {
+            image
+                .frame(width: 80, height: 80)
+                .font(Font.title.weight(.ultraLight))
+                .padding()
+            VStack(alignment: .leading) {
+                HStack {
+                    Text(incident.victimName).bold()
+                    if (incident.victimName != "Name withheld by police") {
+                        Button("Say \(incident.victimGender == "Male" ? "his" : incident.victimGender == "Female" ? "her" : "their") name", action: {
```
Comment (imf) [inline Say Their Names/ContentView.swift:89]: I want to extract this too. Now I have a pattern for that, though... That's what I was struggling with.

File: Say Their Names/UIViewRepresented.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
+//
+//  UIViewRepresented.swift
+//  Say Their Names
+//
+//  Created by Jacob Martin on 7/3/20.
+//
+
+import SwiftUI
+
+struct UIViewRepresented<UIViewType>: UIViewRepresentable where UIViewType: UIView {
```
Comment (imf) [inline Say Their Names/UIViewRepresented.swift:10]: This one you'll have to explain to me. I can feel the goodness, just don't understand it yet.

**Pull request merged**